server/zanata-war/src/test/java/org/zanata/adapter/PropertiesAbstractTest.java

```diff
@@ -0,0 +1,69 @@
+package org.zanata.adapter;
+
+import com.google.common.base.Optional;
+import org.fedorahosted.openprops.Properties;
+import org.zanata.common.ContentState;
+import org.zanata.common.LocaleId;
+import org.zanata.rest.dto.resource.Resource;
+import org.zanata.rest.dto.resource.TextFlowTarget;
+import org.zanata.rest.dto.resource.TranslationsResource;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Damian Jansen <a href="mailto:djansen@redhat.com">djansen@redhat.com</a>
+ */
+class PropertiesAbstractTest {
+
+    String resourcePath = "src/test/resources/org/zanata/adapter/";
+    GenericPropertiesAdapter adapter;
+
+    Resource parseTestFile(String fileName) {
+        File testFile = new File(resourcePath.concat(fileName));
+        assert testFile.exists();
+        return adapter.parseDocumentFile(testFile.toURI(), LocaleId.EN,
+                Optional.absent());
+    }
+
+    TranslationsResource addTranslation(TranslationsResource resource,
+                                        String id, String content, ContentState state) {
+        TextFlowTarget textFlowTarget = new TextFlowTarget();
+        textFlowTarget.setResId(id);
+        textFlowTarget.setContents(content);
+        textFlowTarget.setState(state);
+        resource.getTextFlowTargets().add(textFlowTarget);
+        return resource;
+    }
+
+    File createTempFile(Charset charset) throws Exception {
+        File testFile;
+        try {
+            testFile = File.createTempFile("test-properties-latin1", "properties");
```

Comment:
The filename "test-properties-latin1" could be misleading if `charset` is not Latin 1.